gcp_org_resource_count.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+set -e
+LOG_FILE='gcp_resource_count.log'
+exec 2> >(while IFS= read -r line; do echo "$(date '+%Y-%m-%d %H:%M:%S') [stderr] $line"; done >> $LOG_FILE)
+
+WORKLOAD_VM_UNITS=1
+WORKLOAD_FUNCTION_UNITS=50
+WORKLOAD_SERVERLESS_CONTAINER_UNITS=10
+WORKLOAD_VM_IMAGE_UNITS=1
+WORKLOAD_CONTAINER_IMAGE_UNITS=10
+WORKLOAD_CONTAINER_HOST_UNITS=1
+
+ORG_ID=""
+
+while [[ $# -gt 0 ]]
+do
+    key="$1"
+    case $key in
+        -g|--organization_id)
+        ORG_ID="$2"
+        shift # past argument
+        shift # past value
+        ;;
+        *)    # unknown option
+        shift # past argument
+        ;;
+    esac
+done
+
+if [ -z "$ORG_ID" ]; then
+    echo "[+] Getting organizations list"
+    ORGS=$(gcloud organizations list --format="value(name)")
+    ORGS_LEN=$(echo "$ORGS" | wc -l)
+    echo "Found ${ORGS_LEN} organizations"
+    echo "[+] Iterating organizations and counting resources"
+else
+    ORGS="$ORG_ID"
+    echo "[+] counting resources for organizations id ${ORG_ID}"
+fi
+echo
+
+counter=0
+total_vms=0
+total_functions=0
+total_cloud_run=0
+total_container_images=0
+total_vm_images=0
+total_gke_nodes=0
+total_buckets=0
+total_private_buckets=0
+active_user=$(gcloud config get-value account)
+
+# listed here https://cloud.google.com/asset-inventory/docs/asset-types
+asset_types="compute.googleapis.com/Instance,cloudfunctions.googleapis.com/CloudFunction,run.googleapis.com/Service,containerregistry.googleapis.com/Image,compute.googleapis.com/Image,k8s.io/Node,storage.googleapis.com/Bucket"
+
+for org_id in $ORGS; do
+  echo "Processing organization: $org_id"
+
+  all_resources=$(gcloud asset search-all-resources --scope="organizations/$org_id" --format="json(displayName, assetType, project)" --asset-types="$asset_types")
```

Comment:
can we please have some error handling here? i.e. if one org fails it still makes sense to proceed with other